app/src/main/java/com/randomappsinc/studentpicker/editing/DuplicationDialog.java

```diff
@@ -11,35 +11,41 @@
 public class DuplicationDialog {
 
     public interface Listener {
-        void onDuplicationSubmitted(String name, int amountToAdd);
+        void onDuplicationSubmitted(String name, int amountToHave, int currentAmount);
     }
 
     private MaterialDialog dialog;
     private String currentName;
+    private int currentAmount;
 
     public DuplicationDialog(Context context, final Listener listener) {
         dialog = new MaterialDialog.Builder(context)
                 .content(R.string.cloning_title)
                 .inputType(InputType.TYPE_CLASS_NUMBER)
                 .input(context.getString(R.string.num_copies), "", (dialog, input) -> {
-                    boolean isValid = input.length() > 0 && (Integer.parseInt(input.toString()) > 0);
-                    dialog.getActionButton(DialogAction.POSITIVE).setEnabled(isValid);
+                    if (input.length() > 0) {
+                        int numCopies = Integer.parseInt(input.toString());
+                        boolean isValid = numCopies > 0 && numCopies != currentAmount;
+                        dialog.getActionButton(DialogAction.POSITIVE).setEnabled(isValid);
```

Comment:
I don't think we should limit the user even though it's a weird case. If the user sets the amount to 0, just let them delete all the names.